django/db/backends/base/schema.py

```diff
@@ -996,17 +996,23 @@ def _create_index_name(self, table_name, column_names, suffix=""):
         _, table_name = split_identifier(table_name)
         hash_suffix_part = '%s%s' % (names_digest(table_name, *column_names, length=8), suffix)
         max_length = self.connection.ops.max_name_length() or 200
+        joined_column_names = '_'.join(column_names)
         # If everything fits into max_length, use that name.
-        index_name = '%s_%s_%s' % (table_name, '_'.join(column_names), hash_suffix_part)
-        if len(index_name) <= max_length:
+        index_name = '%s_%s_%s' % (table_name, joined_column_names, hash_suffix_part)
+        if len(index_name.encode('utf-8')) <= max_length:
             return index_name
         # Shorten a long suffix.
         if len(hash_suffix_part) > max_length / 3:
             hash_suffix_part = hash_suffix_part[:max_length // 3]
         other_length = (max_length - len(hash_suffix_part)) // 2 - 1
+        # Shorten column and table names, accounting for multibyte characters.
+        while len(table_name.encode('utf-8')) > other_length:
+            table_name = table_name[:-1]
+        while len(joined_column_names.encode('utf-8')) > other_length:
+            joined_column_names = joined_column_names[:-1]
```

Comment:
In most of cases names won't contain multibyte chars, so it should be worth avoiding multiple encoding and slicing, e.g.:
```python
if len(table_name.encode()) == len(table_name):
    table_name = table_name[:other_length]
else:
    # Shorten table name accounting for multibyte characters.
    while len(table_name.encode()) > other_length:
        table_name = table_name[:-1]
```

Reply:
I added a small hook for this.